src/include/storage/data_table.h

```diff
@@ -281,6 +293,25 @@ class DataTable {
   bool CompareAndSwapVersionPtr(TupleSlot slot, const TupleAccessStrategy &accessor, UndoRecord *expected,
                                 UndoRecord *desired);
 
+  /**
+   * Write a schema message
+   * @param outfile the output file
+   * @param dictionary_ids when encountered columns that are dictionary-encoded, we pre-assign the
+   *                       dictionary id
+   * @param flatbuf_builder flatbuffer builder
+   */
+  void WriteSchemaMessage(std::ofstream &outfile, std::unordered_map<col_id_t, int64_t> *dictionary_ids,
+                          flatbuffers::FlatBufferBuilder *flatbuf_builder) const;
+
+  /**
+   * Write a dictionary message
```

Comment:
Assuming its not a ton of metadata, can you add a comment showing the serialized format, similar to how [`ProjectedRows`](https://github.com/cmu-db/terrier/blob/18c62295b42cc6acd06926989318bd2d94a1e416/src/include/storage/projected_row.h#L19-L23) is shown here